eth2/types/src/newtypes.rs

```diff
@@ -0,0 +1,88 @@
+// Many patterns copied from the slot_epoch.rs file in the
```

Comment:
Why is this called newtypes.rs?  You'll see I did a PR - https://github.com/quilt/lighthouse/pull/1/files#diff-97bf59d81db403d9da467c2c12a3e8cf which introduces slot_epoch patterns that are shard specific. You're going to need epoch conversions, etc. so I'd follow those patterns. Still unsure of why this would be called newTypes vs. just slot_epoch as well?

Reply:
called it newtypes because I have more than slot and epoch in there, and I didn't see a compelling reason to give those their own file.  I'll take a look at your PR, but so far no additional stuff is needed beyond what I have there, but may need more later for sure.

Reply:
Yeah - I'd just avoid a generic name like that and just define it as slot_epoch.rs to stay in pattern.

Reply:
Ok, I don't think I'm totally sold but I don't feel too strongly. Changed to `slot_epoch_root`, happy to change more in the future as necessary.